runtimes/alibi-explain/mlserver_alibi_explain/explainers/sklearn_api_runtime.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any
+
+import joblib
+import lightgbm as lgb
+from xgboost.core import XGBoostError
+from lightgbm.basic import LightGBMError
+
+from mlserver_xgboost.xgboost import _load_sklearn_interface as load_xgb_model
+from mlserver.errors import InvalidModelURI
+from mlserver_alibi_explain.explainers.white_box_runtime import AlibiExplainWhiteBoxRuntime
+
+
+class AlibiExplainSKLearnAPIRuntime(AlibiExplainWhiteBoxRuntime):
+    """
+    Runtime for white-box explainers that require access to a tree-based model matching the SKLearn API, such as
+    a sklearn, XGBoost, or LightGBM model. Example explainers include TreeShap and TreePartialDependence.
+    """
+    async def _get_inference_model(self) -> Any:
+        inference_model_path = self.alibi_explain_settings.infer_uri
+        # Attempt to load model in order: XGBoost, LightGBM, sklearn
+        # TODO - add support for CatBoost (would require model_type = 'classifier' or 'regressor' in settings)
+        try:
```

Comment:
This is pretty ugly, but can't see an obvious alternative unless we add `model_flavour` and `model_type` options to `AlibiExplainSettings`?

Reply:
haha I agree... when you see it written out is a bit weird. 

Happy to add a `model_type` field to the settings? 

A cleaner alternative IMO would be to expect models to provide their own `model-settings.json`. We could then just load the models as usual with their relevant runtimes (and then adding some accessor to retrieve the actual model instance). That way, framework-specific stuff would just fall onto each's runtime implementation. However, assuming that the model's `model-settings.json` will always be present may be too big of an assumption?

Reply:
Discussed in https://github.com/SeldonIO/MLServer/pull/1279#discussion_r1252798610. Decision to stick with `try/except` strategy for now.